### ISSUE TITLE
BUG: generator does not properly escape field with special character $ 

### ISSUE THREAD

**Issue by jtzell** [isar/isar#206]
**Steps to Reproduce**
**Case 1:**
```
class Model {
  int? id;
  String? $fieldWSpecialCharacter;
}
```
Build the above model will produce error extension code.

**Case 2**
```
class Model {
  int? id;
  @Ignore()
  String? $fieldWSpecialCharacter;
}
```
Build the above model will STILL produce error extension code., since `$fieldWSpecialCharacte` is not ignore at all.

**Version**
 - Platform: iOS, Android, Mac, Windows, Linux, Web
 - Flutter version: 2.8.1
 - Isar version: 2.0.0
 - Isar Generator version: 2.0.0

**Comment by leisim** [comment]
Ignore works now with `2.1.0`. Working on special chars!

**Comment by leisim** [comment]
Fixed in `2.1.4` :+1:
